src/parsing/text_reader.rs

```diff
@@ -1,135 +0,0 @@
-    inner: R,
-    decoder: Decoder,
-    eof: bool,
-}
-
-impl<R> TextReader<R>
-where
-    R: BufRead,
-{
-    /// Create a new `TextReader` with the given charset.
-    pub fn new(inner: R, charset: Charset) -> TextReader<R> {
-        TextReader {
-            inner,
-            decoder: charset.new_decoder(),
-            eof: false,
-        }
-    }
-}
-
-impl<R> fmt::Debug for TextReader<R>
-where
-    R: fmt::Debug + BufRead,
-{
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("TextReader")
-            .field("inner", &self.inner)
-            .field("decoder", &"<Decoder>")
-            .field("eof", &self.eof)
-            .finish()
-    }
-}
-
-impl<R> Read for TextReader<R>
-where
-    R: BufRead,
-{
-    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        if self.eof {
-            return Ok(0);
-        }
-
-        let mut total_written = 0;
-
-        loop {
-            let src = self.inner.fill_buf()?;
-            if src.is_empty() {
-                // inner has reached EOF, write last to the buffer.
-                let (res, _, written, _) = self.decoder.decode_to_utf8(src, buf, true);
-                total_written += written;
-
-                match res {
-                    CoderResult::InputEmpty => {
-                        // last call was successful, set eof to true
-                        self.eof = true;
-                        break;
-                    }
-                    CoderResult::OutputFull => {
-                        // last call was not successful because the output is full, try again in the next call to `read`
-                        break;
-                    }
-                }
-            } else {
-                let (res, read, written, _) = self.decoder.decode_to_utf8(src, buf, false);
-
-                self.inner.consume(read);
-                total_written += written;
-
-                match res {
-                    CoderResult::InputEmpty => {
-                        // read all the bytes available in src, read more
-                        continue;
-                    }
-                    CoderResult::OutputFull => {
-                        // buf is full, break and return the number read
-                        break;
-                    }
-                }
-            }
-        }
-
-        Ok(total_written)
-    }
-}
-
-#[test]
-fn test_stream_decoder_utf8() {
-    let mut reader = TextReader::new("québec".as_bytes(), crate::charsets::UTF_8);
-
-    let mut text = String::new();
-    assert_eq!(reader.read_to_string(&mut text).ok(), Some(7));
-
-    assert_eq!(text, "québec");
-}
-
-#[test]
-fn test_stream_decoder_latin1() {
-    let mut reader = TextReader::new(&b"qu\xC9bec"[..], crate::charsets::WINDOWS_1252);
-
-    let mut text = String::new();
-    assert_eq!(reader.read_to_string(&mut text).ok(), Some(7));
-
-    assert_eq!(text, "quÉbec");
-}
-
-#[test]
-fn test_string_reader_large_buffer_latin1() {
```

Comment:
Can you leave this test in just to make sure it really does what we want?

Reply:
Also this test was buggy, it did not assert that the length read to the string was at least 10,000 which would have caught the bug earlier on.

Reply:
Will do when I add back a `TextReader` type.

Reply:
Done.